fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -300,7 +294,27 @@ ThreadResult<RangeResult> DLTransaction::readBlobGranules(const KeyRangeRef& key
 	                                                         beginVersion,
 	                                                         rv,
 	                                                         context);
-	return ThreadResult<RangeResult>((ThreadSingleAssignmentVar<RangeResult>*)(r));
+	const FdbCApi::FDBKeyValue* kvs;
+	int count;
+	FdbCApi::fdb_bool_t more;
+	FdbCApi::fdb_error_t error = api->resultGetKeyValueArray(r, &kvs, &count, &more);
+	if (error) {
+		return ThreadResult<RangeResult>(Error(error));
```

Comment:
I still think this case needs to call resultDestroy on r before returning, otherwise it will leak r's TSAV?
Running some ASAN tests would confirm this, if they hit errors.

Reply:
If we call resultDestroy on r before returning, then that would invalidate the memory referenced in the result. We want to defer calling resultDestroy until the lifetime of the RangeResult's arena ends.

Reply:
But for the error case, we are just returning the error code from the result, right? None of the memory in the result matters besides that,and we copy it into the "error" field.
It also might not be hitting the error case all of the time in in fdb_c_api_tests_blob_granule

Reply:
Good point. I didn't consider this. Updated, thanks!